src/server/ua_server_discovery_mdns.c

```diff
@@ -615,10 +495,14 @@ mdns_create_txt(UA_Server *server, const char *fullServiceDomain, const char *pa
         xht_set(h, "path", "/");
     } else {
         /* path does not contain slash, so add it here */
-        if(path[0] == '/')
-            /* todo: malloc in strdup may fail: return a statuscode */
-            allocPath = UA_STRDUP(path);
-        else {
+        if(path[0] == '/') {
+            allocPath = (char*)UA_malloc(sizeof(path));
```

Comment:
Is sizeof intended here? A few lines later strlen(path) [+ 1] is used for this.

Reply:
Oooops, no, should be strlen.... Thanks